### PR TITLE
Newsletters: add 310 (2024-07-05)

### PR DESCRIPTION
**Pull request by harding**
- [x] Lede, releases/RCs, topic entries @harding 
- [x] Code changes @Gustavojfe

### CODE REVIEW

File: _posts/en/newsletters/2024-07-05-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+- **Adding a BOLT11 invoice field for blinded paths:** Elle Mouton
+  [posted][mouton b11b] to Delving Bitcoin a proposed BLIP specification
+  for an optional field that could be added to [BOLT11][] invoices to
+  communicate a [blinded path][topic rv routing] for paying the
+  receiver's node.  For example, businessperson Bob wants to receive a
+  payment from customer Alice but does not want to disclose the identity
+  of his node or of the peers with whom he shares channels.  He
+  generates a blinded path starting a few hops out from his node and
+  adds that to an otherwise-standard BOLT11 invoice that he gives Alice.
+  If Alice uses software that is able to parse that invoice and route a
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-07-05-newsletter.md:23]:
```suggestion
  If Alice uses software that can parse that invoice and route a
```

File: _posts/en/newsletters/2024-07-05-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  payment using blinded paths, she can pay Bob.
+
+  If Alice uses software which doesn't support the BLIP, she will be
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-07-05-newsletter.md:26]:
```suggestion
  If Alice uses software that doesn't support the BLIP, she will be
```

**Comment by harding** [comment]
FYI: I'm working on adding a news item about this for this week's newsletter: https://mailing-list.bitcoindevs.xyz/bitcoindev/xsylfaVvODFtrvkaPyXh0mIc64DWMCchxiVdTApFqJ_0Q5v0bOoDpS_36HwDKmzdDO9U2RKMzESEiVaq47FTamegi2kCNtVZeDAjSR4G7Ic=@protonmail.com/T/#u

**Comment by Gustavojfe** [comment]
just pushed all the summaries of the week. thank you for the review and feedback.

File: _posts/en/newsletters/2024-07-05-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+    require significant effort to create bogus headers starting from it.
+    However, a code change incorporated in Bitcoin Core 0.12 allowed the
+    node to begin accepting low-difficulty headers into memory,
+    potentially allowing an attacker to fill memory with bogus headers.
+    This could lead to the node crashing, which could result in the loss
+    of funds for users of contract protocols (such as LN).  Cory Fields
+    responsibly disclosed the vulnerability and it was fixed in version
+    0.15.0.
+
+  - [CPU-wasting DoS due to malformed requests][]: before Bitcoin
```
Comment (jirijakes) [inline _posts/en/newsletters/2024-07-05-newsletter.md:158]:
```suggestion
  - [CPU-wasting DoS due to malformed requests][]: before Bitcoin Core
```

**Review by jirijakes — COMMENTED** [review]
Reviewed disclosures while translating. Excellent summary!

File: _posts/en/newsletters/2024-07-05-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+- [BIPs #1610][] adds [BIP379][] with a specification for [Miniscript][topic
+  miniscript], a language that compiles to Bitcoin Script but which allows
+  composition, templating, and definitive analysis. See [Newsletter #304][news304
+  miniscript] for an earlier reference to this BIP.
+
+- [BIPs #1540][] adds BIPs [328][bip328], [390][bip390], and [373][bip373] with a specification for a
+  [MuSig2][topic musig] derivation scheme for aggregate keys (328), output
+  script [descriptors][topic descriptors] (390), and [PSBT][topic psbt] fields
+  to allow MuSig2 data to be included in a PSBT of any version (373). MuSig2 is
+  a protocol for aggregating public keys and signatures for the [Schnorr digital
```
Comment (jirijakes) [inline _posts/en/newsletters/2024-07-05-newsletter.md:298]: Optech tends to use lowercase “schnorr signature”.

```suggestion
  a protocol for aggregating public keys and signatures for the [schnorr digital
```

**Review by jirijakes — COMMENTED** [review]
I like this more concise format of notable changes. Great job, @Gustavojfe!

File: _posts/en/newsletters/2024-07-05-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+affecting old versions of Bitcoin Core and describes a proposal to allow
+BOLT11 invoices to include blinded paths.  Also included are our regular
+sections announcing new releases and release candidates and summarizing
+notable changes to popular Bitcoin infrastructure software.
+
+## News
+
+- **Disclosure of vulnerabilities affecting Bitcoin Core versions before 0.21.0:**
+  Antoine Poinsot [posted][poinsot disclose] to the Bitcoin-Dev mailing
+  list a link to an [announcement][bcco announce] of 10 vulnerabilites
```
Comment (azuchi) [inline _posts/en/newsletters/2024-07-05-newsletter.md:20]:
```suggestion
  list a link to an [announcement][bcco announce] of 10 vulnerabilities
```

**Comment by harding** [comment]
Made edits for all feedback (thanks @LarryRuane @jirijakes @azuchi), reviewed and edited @Gustavojfe section (thanks!), added lede, releases/RCs (no changes from initial draft), and topic entries.  Thanks everyone!

**Review by bitschmidty — COMMENTED** [review]
Sorry for the late review. I pushed commits for a couple issues I noticed.

**Pull request merged**